eslint.config.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import jsdoc from 'eslint-plugin-jsdoc';
+import js from '@eslint/js';
+import {FlatCompat} from '@eslint/eslintrc';
+import babelParser from '@babel/eslint-parser';
+import globals from 'globals';
+import tsParser from '@typescript-eslint/parser';
+import typescript from '@typescript-eslint/eslint-plugin';
+
+// Used to convert the Google eslint config into flat config format.
+const compat = new FlatCompat();
+
+export default [
```

Comment:
I purposefully didn't comment on any of the rules here because I don't want to bog this formatting PR down in an ESLint discussion :P

Seems to be the same as what we currently have, including the existing differences from core. So I think this is good for now! If there's anything you purposefully changed that you want to call out happy to discuss.

Reply:
Yeah, the only things I think I changed were disabling the fixers for missing jsdoc because it kept creating empty comments, and disabling the `tag-lines` rule because there were a lot of violations and it was clogging my ability to make sure I had everything else correct.